test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -0,0 +1,478 @@
+from Standard.Base import all
+import Standard.Base.Data.Numeric.Internal.Numbers_Constants
+import Standard.Base.Errors.Common.Arithmetic_Error
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+
+from Standard.Base.Data.Numbers import Number_Parse_Error
+from Standard.Base.Errors.Common import Loss_Of_Numeric_Precision
+
+from Standard.Test import all
+
+polyglot java import org.enso.base.numeric.Decimal_Utils
+
+Decimal.should_have_rep self rep = self.internal_representation . should_equal rep
+
+add_specs suite_builder =
+    suite_builder.group "construction" group_builder->
+        group_builder.specify "should be able to construct a Decimal from a string" <|
+            Decimal.new "123.45" . should_have_rep [12345, 5, 2]
+            Decimal.parse "123.45" . should_have_rep [12345, 5, 2]
+
+        group_builder.specify "should throw Number_Parse_Error on a badly-formatted string" <|
+            Decimal.new "ee" . should_fail_with Number_Parse_Error
+            Decimal.new "--123.3.3" . should_fail_with Number_Parse_Error
+
+        group_builder.specify "should be able to construct a Decimal from a string, with an explicit precision, using the default rounding" <|
+            mc = Math_Context.new 4
+            Decimal.new "123.45" mc . should_have_rep [1235, 4, 1]
+            Decimal.new "123.45" mc . should_equal (Decimal.new "123.5")
+
+        group_builder.specify "should be able to construct a Decimal from a string, with an explicit precision, using a non-default rounding" <|
+            mc = Math_Context.new 4 Rounding_Mode.bankers
+            Decimal.new "123.45" mc . should_have_rep [1234, 4, 1]
+            Decimal.new "123.45" mc . should_equal (Decimal.new "123.4")
+
+        group_builder.specify "should be able to construct a Decimal from a long string" <|
+            Decimal.new "495782984723948723947239938732974241.2345" . should_have_rep [4957829847239487239472399387329742412345, 40, 4]
+            Decimal.parse "495782984723948723947239938732974241.2345" . should_have_rep [4957829847239487239472399387329742412345, 40, 4]
+
+        group_builder.specify "should be able to construct a Decimal from a small integer" <|
+            Decimal.new 1234500 . should_have_rep [1234500, 7, 0]
+
+        group_builder.specify "should be able to construct a Decimal from a small integer, with an explicit precision, using the default rounding" <|
+            mc = Math_Context.new 4
+            Decimal.new 1234500 mc . should_have_rep [1235, 4, -3]
+            Decimal.new 1234500 mc . should_equal (Decimal.new 1235000)
+
+        group_builder.specify "should be able to construct a Decimal from a small integer, with an explicit precision, using the non-default rounding" <|
+            mc = Math_Context.new 4 Rounding_Mode.bankers
+            Decimal.new 1234500 mc . should_have_rep [1234, 4, -3]
+            Decimal.new 1234500 mc . should_equal (Decimal.new 1234000)
```

Comment:
Let's add precision loss checks for these cases as well.

Because if I see correctly tests only validate the Float case.

Reply:
The "should report precision loss accurately" test does this for all three types. Is that what you mean?

Reply:
Yes. I was looking at it and somehow did not get it, but it looks alright on second look.